### PR TITLE
Introduced paramsToIgnore

### PR DESCRIPTION
**Pull request by OleksiiBaiev-tomtom**
Introduced using paramsToIgnore

### CODE REVIEW

File: DVR.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -860,6 +860,7 @@
 				DYLIB_INSTALL_NAME_BASE = "@rpath";
 				INFOPLIST_FILE = "$(SRCROOT)/Sources/DVR/Resources/Info.plist";
 				INSTALL_PATH = "$(LOCAL_LIBRARY_DIR)/Frameworks";
+				IPHONEOS_DEPLOYMENT_TARGET = 13.0;
```
Comment (YuryLapitsky-TomTom) [inline DVR.xcodeproj/project.pbxproj]: let's revert this one to keep fork compatible with the original repo

File: Sources/DVR/Cassette.swift [hunk 2]
```diff
@@ -83,4 +83,27 @@ private extension URLRequest {
         }
         return true
     }
+    
+    func hasParamsEqualToThatOfRequest(_ request: URLRequest, ignoreParams: [String] = []) -> Bool {
```
Comment (YuryLapitsky-TomTom) [inline Sources/DVR/Cassette.swift]:
```suggestion
    func hasEqualParameters(for request: URLRequest, ignoreParameters: [String] = []) -> Bool {
```

File: Sources/DVR/Cassette.swift [hunk 3]
```diff
@@ -83,4 +83,27 @@ private extension URLRequest {
...
+        if url == request.url { return true }
+            
+        let request1 = createRequest(withoutKeys: ignoreParams)
+        let request2 = request.createRequest(withoutKeys: ignoreParams)
+        
+        if request1.url == request2.url { return true }
+        
+        return false
```
Comment (YuryLapitsky-TomTom) [inline Sources/DVR/Cassette.swift]:
```suggestion
        return request1.url == request2.url
```

File: Sources/DVR/Cassette.swift [hunk 4]
```diff
@@ -83,4 +83,27 @@ private extension URLRequest {
         }
         return true
     }
+    
+    func hasParamsEqualToThatOfRequest(_ request: URLRequest, ignoreParams: [String] = []) -> Bool {
+        if url == request.url { return true }
```
Comment (YuryLapitsky-TomTom) [inline Sources/DVR/Cassette.swift:88]:
```suggestion
        guard url != request.url else { return true }
```

File: Sources/DVR/Interaction.swift [hunk 5]
```diff
@@ -101,7 +101,7 @@ extension Interaction {
         guard let request = dictionary["request"] as? [String: Any],
             let response = dictionary["response"] as? [String: Any],
             let recordedAt = dictionary["recorded_at"] as? TimeInterval else { return nil }
-
+
```
Comment (YuryLapitsky-TomTom) [inline Sources/DVR/Interaction.swift]: unnecessary changes

File: Sources/DVR/Session.swift [hunk 6]
```diff
@@ -15,6 +15,7 @@ open class Session: URLSession {
 
     private let testBundle: Bundle
     private let headersToCheck: [String]
+    private let paramsToIgnore: [String]
```
Comment (YuryLapitsky-TomTom) [inline Sources/DVR/Session.swift]:
```suggestion
    private let parametersToIgnore: [String]
```

File: Sources/DVR/Session.swift [hunk 7]
```diff
@@ -28,12 +29,18 @@ open class Session: URLSession {
 
     // MARK: - Initializers
 
-    public init(outputDirectory: String = "~/Desktop/DVR/", cassetteName: String, testBundle: Bundle = Session.defaultTestBundle!, backingSession: URLSession = URLSession.shared, headersToCheck: [String] = []) {
+    public init(outputDirectory: String = "~/Desktop/DVR/", 
+                cassetteName: String,
+                testBundle: Bundle = Session.defaultTestBundle!,
+                backingSession: URLSession = URLSession.shared,
+                headersToCheck: [String] = [],
+                paramsToIgnore: [String] = []) {
```
Comment (YuryLapitsky-TomTom) [inline Sources/DVR/Session.swift]:
```suggestion
                parametersToIgnore: [String] = []) {
```

File: DVR.xcodeproj/project.pbxproj (hunk 1 above)
Comment (OleksiiBaiev-tomtom) [reply DVR.xcodeproj/project.pbxproj]: I will do but now it needs to run tests

File: Sources/DVR/Cassette.swift (hunk 2 above)
Comment (OleksiiBaiev-tomtom) [reply Sources/DVR/Cassette.swift]: Done

File: Sources/DVR/Cassette.swift (hunk 4 above)
Comment (OleksiiBaiev-tomtom) [reply Sources/DVR/Cassette.swift:88]: Done

File: Sources/DVR/Interaction.swift (hunk 5 above)
Comment (OleksiiBaiev-tomtom) [reply Sources/DVR/Interaction.swift]: Done

File: Sources/DVR/Session.swift (hunk 6 above)
Comment (OleksiiBaiev-tomtom) [reply Sources/DVR/Session.swift]: Done

File: Sources/DVR/Session.swift (hunk 7 above)
Comment (OleksiiBaiev-tomtom) [reply Sources/DVR/Session.swift]: Done

File: Sources/DVR/Cassette.swift (hunk 3 above)
Comment (OleksiiBaiev-tomtom) [reply Sources/DVR/Cassette.swift]: Done

File: Tests/DVRTests/SessionTests.swift [hunk 8]
```diff
@@ -269,4 +314,27 @@ class SessionTests: XCTestCase {
 
         wait(for: [firstExpectation, secondExpectation], timeout: 3.0)
     }
+        
+    func testEVSearch() throws {
+        let session = Session(cassetteName: "EVSearch", parametersToIgnore: ["key"])
+        
+        let url = try XCTUnwrap  (URL(string:"https://api.tomtom.com/search/2/evsearch?radius=5000&lon=4.909466&limit=2&lat=52.377271&key=INSERT_IN_RECORDING&connectorSet=&accessTypes=Public"))
```
Comment (YuryLapitsky-TomTom) [inline Tests/DVRTests/SessionTests.swift]: https://httpbin.org/response-headers?apiKey=val&format=json

File: Tests/DVRTests/SessionTests.swift [hunk 9]
```diff
@@ -108,6 +108,30 @@ class SessionTests: XCTestCase {
 
         waitForExpectations(timeout: 1, handler: nil)
     }
+    
+    func testTextPlaybackWithParams() {
```
Comment (YuryLapitsky-TomTom) [inline Tests/DVRTests/SessionTests.swift]: you can send 2 requests, one with a key another one without a key. And they should succeed and provide the same response

File: Tests/DVRTests/SessionTests.swift [hunk 10]
```diff
@@ -235,6 +259,27 @@ class SessionTests: XCTestCase {
 
         waitForExpectations(timeout: 1, handler: nil)
     }
+    
+    func testRecordingWithIgnoredParams() {
```
Comment (YuryLapitsky-TomTom) [inline Tests/DVRTests/SessionTests.swift]: let's remove this one as recording crashes the tests

File: Tests/DVRTests/SessionTests.swift [hunk 11]
```diff
@@ -269,4 +314,27 @@ class SessionTests: XCTestCase {
 
         wait(for: [firstExpectation, secondExpectation], timeout: 3.0)
     }
+        
+    func testEVSearch() throws {
```
Comment (YuryLapitsky-TomTom) [inline Tests/DVRTests/SessionTests.swift]: I think this one duplicates the coverage provided in `testTextPlaybackWithParams`, so probably we don't need the second test of the same type.

**Review by YuryLapitsky-TomTom — APPROVED** [review]

File: Tests/DVRTests/SessionTests.swift (hunk 10 above)
Comment (OleksiiBaiev-tomtom) [reply Tests/DVRTests/SessionTests.swift]: Removed

File: Tests/DVRTests/SessionTests.swift (hunk 11 above)
Comment (OleksiiBaiev-tomtom) [reply Tests/DVRTests/SessionTests.swift]: Removed

File: Tests/DVRTests/SessionTests.swift (hunk 8 above)
Comment (OleksiiBaiev-tomtom) [reply Tests/DVRTests/SessionTests.swift]: Done

File: Tests/DVRTests/SessionTests.swift (hunk 9 above)
Comment (OleksiiBaiev-tomtom) [reply Tests/DVRTests/SessionTests.swift]: Done
